text/0000-re-org-docs.md

```diff
@@ -0,0 +1,291 @@
+│   │   │   ├── create-builder.svg
+│   │   │   ├── extensions
+│   │   │   │   └── extensions.md
+│   │   │   ├── lifecycle
+│   │   │   │   ├── _index.md
+│   │   │   │   ├── analyze.md
+│   │   │   │   ├── build.md
+│   │   │   │   ├── create.md
+│   │   │   │   ├── detect.md
+│   │   │   │   ├── export.md
+│   │   │   │   ├── launch.md
+│   │   │   │   ├── rebase.md
+│   │   │   │   └── restore.md
+│   │   │   └── metadata
+│   │   ├── operations
+│   │   │   ├── _index.md
+│   │   │   ├── build.md
+│   │   │   ├── build.svg
+│   │   │   ├── rebase.md
+│   │   │   └── rebase.svg
+│   │   └── platform.md
+│   ├── how-to-guides
+│   │   ├── create-a-builder.md
+│   │   ├── create-a-stack.md
+│   │   ├── migrate
+│   │   │   ├── _index.md
+│   │   │   ├── deprecated
+│   │   │   │   ├── platform-api-0.3-0.4.md
+│   │   │   │   ├── platform-api-0.4-0.5.md
+│   │   │   │   ├── platform-api-0.5-0.6.md
+│   │   │   │   └── platform-api-0.6-0.7.md
+│   │   │   ├── platform-api-0.10-0.11.md
+│   │   │   ├── platform-api-0.11-0.12.md
+│   │   │   ├── platform-api-0.7-0.8.md
+│   │   │   ├── platform-api-0.8-0.9.md
+│   │   │   └── platform-api-0.9-0.10.md
+│   │   ├── specify-extensions.md
+│   │   └── integrate-ci
+│   │       ├── circleci.md
+│   │       ├── gitlab.md
+│   │       ├── kpack.md
+│   │       ├── pack
+│   │       │   ├── cli
+│   │       │   └── concepts
+│   │       │       └── trusted_builders.md
+│   │       ├── piper.md
+│   │       └── tekton.md
+│   └── tutorials <- NOTE: we should add stuff here
+└── reference <- NOTE: we should keep this content as lean as possible and link back to the spec where we can, as it tends to get out of date
+│   ├── config
+│   │   ├── builder-config.md
+│   │   ├── package-config.md
+│   │   └── project-descriptor.md
+│   └── spec
+│   │   ├── buildpack-api.md
+│   │   ├── distribution-api.md
+│   │   └── platform-api.md
+```
+
+# Migration
+[migration]: #migration
+
+A lot of content would be moving from the buildpack tutorial to the buildpack how-to section.
+The current buildpack tutorial has the end-user make a `ruby` buildpack by editing the same file over and over.
+It is good for teaching but hard to explain advanced concepts like multiple processes, SBOM, etc.
+We should break the advanced concepts out of the tutorial and link back to the samples (or just give short code snippets) instead.
+
+As suggested by @AidanDelaney, we should consider changing the tutorial to make a `nodejs` buildpack:
+
+> The current example installs the runtime, installs the package manager, and resolves the packages. Could we simply install a nodejs runtime and build a vanilla application?
+
+We should make these changes in one huge PR. We should NOT attempt to redirect from old links as that would be too hard.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+- It's a lot of work
+- Changing the structure of the docs could be surprising for users who are already familiar with its content
```

Comment:
It also potentially breaks SEO and links to the documentation outside of us. Will we redirect to the homepage or other pages?